courses/tests/test_fields.py

```diff
@@ -0,0 +1,95 @@
+"""
+Tests for Serializer Fields
+"""
+from django.core.exceptions import ImproperlyConfigured
+from django.test import TestCase
+
+from rest_framework.serializers import ValidationError
+
+from courses.factories import EdxAuthorFactory, CourseFactory
+from courses.models import EdxAuthor
+from courses.serializers import JsonListField as JLF
+from courses.serializers import StringyManyToManyField as SMMF
+
+
+class JsonListFieldTests(TestCase):
+    """
+    Tests for JsonListField
+    """
+    def test_decodes_string(self):
+        """
+        Test that empty list string decodes properly
+        """
+        f = JLF()
+        self.assertEquals([], f.to_internal_value('[]'))
+
+    def test_decodes_unicode(self):
+        """
+        Test that empty list unicode string decodes properly
+        """
+        f = JLF()
+        self.assertEquals([], f.to_internal_value(u'[]'))
+
+    def test_handles_decoding_nullable_values(self):
+        """
+        Test that null is decoded to None
+        """
+        f = JLF()
+        self.assertEquals(None, f.to_internal_value('null'))
+
+    def test_throws_validationerror_on_invalid_json(self):
+        """
+        Test invalid JSON
+        """
+        f = JLF()
+        self.assertRaises(ValidationError, f.to_internal_value, 'testing')
+
+    def test_not_list(self):
+        """
+        Test that to_internal_value takes only lists
+        """
+        f = JLF()
+        self.assertRaises(ValidationError, f.to_internal_value, '{}')
+
+
+class StringyM2MTestCase(TestCase):
+    """Tests for m2m stringy field serializer"""
+    def test_requires_model(self):
+        """Field requires a model kwarg"""
+        self.assertRaises(ImproperlyConfigured, SMMF, lookup='test')
+
+    def test_requires_lookup(self):
+        """Field requires a lookup kwarg"""
+        self.assertRaises(ImproperlyConfigured, SMMF, model=EdxAuthor)
+
+    def test_returns_string_for_all_objects(self):  # pylint: disable=no-self-use
+        """model-to-string returns correct strings"""
+        e1 = EdxAuthorFactory.create()
+        e2 = EdxAuthorFactory.create()
+        co = CourseFactory.create()
+        co.instructors.add(e1)
+        co.instructors.add(e2)
+
+        f = SMMF(model=EdxAuthor, lookup='edx_uid')
+        assert sorted([str(e1), str(e2)]) == sorted(f.to_representation(co.instructors))
```

Comment:
These assertions result in dramatically better output for py.test. Py.test completely rewrites this code. See here: https://pytest.org/latest/assert.html